force-app/main/default/lwc/paginator/__tests__/paginator.test.js

```diff
@@ -0,0 +1,77 @@
+import { createElement } from 'lwc';
+import Paginator from 'c/paginator';
+
+describe('c-paginator', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+    });
+
+    it('sends "next" and "previous" events on button click', () => {
+        // Create initial element
+        const element = createElement('c-paginator', {
+            is: Paginator
+        });
+        document.body.appendChild(element);
+
+        // Mock handlers for child events
+        const handlerPrevious = jest.fn();
+        const handlerNext = jest.fn();
+        // Add event listener to catch child events
+        element.addEventListener('previous', handlerPrevious);
+        element.addEventListener('next', handlerNext);
+
+        element.shadowRoot
+            .querySelectorAll('lightning-button-icon')
```

Comment:
BTW this is not a good way to test: clicking all buttons at once won't tell you if previous does indeed go back.
Test would still pass if button were doing the opposite of what label says.

Reply:
Implemented with changes

Reply:
This is somehow returning null. I will check and fix it to select only one element.